M2/Macaulay2/d/interface.dd

```diff
@@ -7,30 +7,40 @@ use common;
 use hashtables;
 use struct;
 
-header "
-  #include <engine.h>
-  #include <ZZp.hpp>
-  #include <monomial.hpp>
-  #include <exceptions.hpp>
-  #include <relem.hpp>
-  #include <aring-glue.hpp>
-  #include <aring-zzp.hpp>
-
-  #include <stdexcept>
-  void test_catch0() { 
-       throw (std::runtime_error(\"test\")); 
-       }
-  int test_catch() {
-       try { test_catch0(); } catch (const std::runtime_error& x) { return 1; }
-       return 0;
-       }
-  
-  ";
+header "// TODO: break apart this file so each piece includes only one:
+#include <interface/aring.h>
+#include <interface/cra.h>
+#include <interface/factory.h>
+#include <interface/flint.h>
+#include <interface/freemodule.h>
+#include <interface/groebner.h>
+#include <interface/matrix.h>
+#include <interface/monoid.h>
+#include <interface/monomial-ideal.h>
+#include <interface/monomial-ordering.h>
+#include <interface/mutable-matrix.h>
+#include <interface/random.h>
+#include <interface/ring.h>
+#include <interface/ringelement.h>
+#include <interface/ringmap.h>";
+
+header "// TODO: remove the following headers
```

Comment:
@mahrud What is the meaning of the "remove the following headers" here?  Are they not used?

Reply:
They are currently used for the symbols mentioned at the end of each line, but we should eventually make sure all of those symbols are exported through one of the interface headers, then we can exclusively include interface headers and remove these non-interface headers.